lib/Dialect/ESI/runtime/cpp/lib/backends/Cosim.cpp

```diff
@@ -394,6 +394,35 @@ class CosimMMIO : public MMIO {
   std::unique_ptr<FuncService::Function> readMMIO;
 };
 
+class CosimHostMem : public HostMem {
+public:
+  CosimHostMem() {}
+
+  struct CosimHostMemRegion : public HostMemRegion {
+    CosimHostMemRegion(std::size_t size) {
+      ptr = malloc(size);
+      this->size = size;
+    }
```

Comment:
Consider makign the constructor private if CosimHostMem is the only thing that should ever create these.

Reply:
It's in a cpp file, so nothing outside the file can construct it anyway.